packages/core/src/interpreter.ts

```diff
@@ -453,6 +464,24 @@ export class Interpreter<
     return this;
   }
 
+  private _transition(
+    state: State<TContext, TEvent>,
+    event: SCXML.Event<TEvent>
+  ) {
+    try {
```

Comment:
What kind of errors can occur here?

Reply:
Caused by invalid arguments, problems in our implementation and probably some errors thrown when we call to the user’s code as I think we dont convert everything correctly to error.execution events.

This is just a cleanup so we dont land with some corrupted internal state - if we did it would mean that some other error has happened before (here) and it should be handled/fixed. In ideal world - this shouldnt be needed. OTOH this cleans avoids an extremely small memory leak in extremely unlikely situation.